proxy/src/bind.rs

```diff
@@ -524,3 +554,22 @@ impl Protocol {
         }
     }
 }
+
+// ===== impl RebindTls =====
+
+impl<B> Rebind<Option<tls::ClientConfig>> for RebindTls<B>
+where
+    B: tower_h2::Body + Send + 'static,
+    <B::Data as ::bytes::IntoBuf>::Buf: Send,
+{
+    type Service = StackInner<B>;
+    fn rebind(&mut self, _cfg: &Option<tls::ClientConfig>) -> Self::Service {
+        debug!(
+            "rebinding endpoint stack for {:?}:{:?} on TLS config change",
+            self.endpoint, self.protocol,
+        );
+        // We don't actually pass in the config, as `self.bind` also already
+        // owns a config watch of its own.
```

Comment:
Can we just pass the config through rather than check the borrow again later? It's needlessly complex.  (Imagine, for instance, that something were to modify cfg before passing it to rebind...)  This is too subtle as it is.

Reply:
I didn't want to modify the `bind_stack` signature to take the config, because then other callers of `bind_stack` (i.e. `BoundService`) would also have to have a config to pass in.

Reply:
Let's not make this a blocker, but this is a problem that needs to be fixed, whether by changing the bind_stack API or doing something else

Reply:
My intention was to come back and remove the client config watch from the `ctx::Proxy` context (but I didn't do that here as it seemed out of scope for this branch). When I do that, my intention was to move the watch to an explicit field on `Bind`. Once _that_'s done, I'd expect `bind` to borrow _that_ config, so the config watch used by `WatchService` is, in this case, only used for the notification that the watch has changed.

Reply:
> Once that's done, I'd expect bind to borrow that config, so the config watch used by WatchService is, in this case, only used for the notification that the watch has changed.

I have a problem with this approach.  Either we should make the WatchService take a Watch<()>, or we should use the config that it provides us

Reply:
i.e. if we have an API that is "rebind with this value" and then we throw out that value and use another one, that's a bad API. Rebind either should not take a value (which seems like a poor choice for WatchService, or we should change our use to fit the API better